### ISSUE TITLE
SSH key pair authentication fails while executing ./scripts/setup.sh file

### ISSUE THREAD

**Issue by gewitterblitz** [jbusecke/cookiecutter-science-project#7]
As per the changes log on github website, the curl -u <username> command line feature for ssh access to github has been deprecated. See [here](https://developer.github.com/changes/2020-02-14-deprecating-password-auth/).

When executing the `scripts/setup.sh` bash script, I got the following error:

```
ERROR: Repository not found.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I could solve it using the updated curl -H command after setting up a personal access token through github.

Happy to submit a PR if you wish!

**Comment by jbusecke** [comment]
That would be awesome! Can you make sure to add some instructions on how to do that on the github side? Thanks for working on this.

**Comment by milindsharma1** [comment]
So, this seems a bit more complicated than I thought it to be. 

Adding a new context in the cookiecutter.json file allowing user input for their github personal access tokens should ideally work (we'll also modify the current curl -u command in scripts/setup.sh). However, the problem with that approach is that everyone else can also see user's PAT in case they choose to keep their github repos public (which ofc is not good).

I spent some time looking for alternative approaches but couldn't really find an immediate solution.

Any ideas/thoughts?

**Comment by jbusecke** [comment]
No that is definitely not good, I would not want to create such a security risk.

Is there a way to pipe the access token into the script without ever saving them?

I'll try to look into this more in the coming days, currently very busy with a proposal.

**Comment by wenegrat** [comment]
I'm having the same problem as @gewitterblitz. Do you know if there's any workaround for this at the moment? Thanks!

**Comment by jbusecke** [comment]
I have briefly looked into this, but must admit that I dont feel well versed enough to try to change this by myself (because of security risks mentioned above). I will look into this again, and see if I can somehow get this working in a similar fashion...otherwise we might have to deprecate this and create a gh repo by hand...😩

**Comment by jbusecke** [comment]
Ok I think I found a solution. One can use githubs [Comman Line Interface](https://github.com/cli/cli#installation) to create repos. The authentication has to be done only one and is pretty straight forward (using a browser and authentication code). 
The user would however have to install this first (I used homebrew and it was pretty painless). What do people think about this solution? I am not an expert on security, but since this is made by the GH folks I would not worry about it at this point. 

Also do you all prefer to have this all wrapped into a setup script? Or should we just add instructions how to do it? This might be a bit more clean, since you also would not have to enter as much details during the cookiecutter setup process (e.g. github repo name, private/public repo).

**Comment by milindsharma1** [comment]
I think instructions should be enough for users to follow along.

**Comment by jbusecke** [comment]
I will implement this later today.

**Comment by jbusecke** [comment]
Would love to hear your feedback on #14 @gewitterblitz. Also pinging @wenegrat and @gmacgilchrist.

**Comment by jbusecke** [comment]
Closed via #15
